extensions/Datatype.php

```diff
@@ -0,0 +1,98 @@
+<?php
+
+trait Datatype {
+
+    /**
+     * Datatypes
+     * Datatypes of CSV data-columns
+     *
+     * @access public
+     * @var array
+     */
+    public $data_types = [];
+
+    /**
+     * Check data type
+     * Check for possible data types for one field value string.
+     *
+     * @access private
+     *
+     * @param string $value cell value
+     *
+     * @return string
+     */
+    private function getDatatypeFromString($value) {
+        $value = trim((string) $value);
+
+        if (empty($value)) {
+            return 'unknown';
+        }
+
+        if (preg_match('/^(?i:true|false)$/', $value)) {
+            return 'boolean';
+        }
+
+        if (preg_match('/^[-+]?[0-9]\d*$/', $value)) {
+            return 'integer';
+        }
+
+        if (preg_match('/^[+-]?([0-9]*[.])?([0-9]|[.][0-9])+$/', $value)) {
+            return 'float';
+        }
+
+        if ((bool) strtotime($value)) {
+            return 'date';
+        }
+
+        return 'string';
+    }
+
+    /**
+     * Check data type for one column.
+     * Check for most commonly data type for one column.
+     *
+     * @access private
+     *
+     * @param  [array] $datatypes
+     *
+     * @return [string|false]
```

Comment:
PHPDoc: I like that you've added documentation blocks. Please remove the [] to keep the style compatible to the rest of this project's code.

Reply:
done.